### PR TITLE
FED-2112 Required props lint: respect @Props(ignoreRequiredProps: ...) 

### PR DESCRIPTION
**Pull request by greglittlefield-wf**
## Dependent PR
These changes are dependent on two different PRs:
- https://github.com/Workiva/over_react/pull/871
- https://github.com/Workiva/over_react/pull/874

For review, I've set the base as a branch that merges the two together.

## Motivation
Part of adding `ignoreRequiredProps` functionality is supporting it in the analyzer plugin, by making the required props lint not treat those props as required.

## Changes
- Update `getAllRequiredProps` logic to mark props specified within `@Props(ignoreRequiredProps: ...)` as ignored
    - Add unit tests
    - Add diagnostic integration tests
    - Add playground example
- Add ignore for `invalid_use_of_visible_for_overriding_member` to generated code, since it was causing analyzer plugin test setups to fail when verifying that there are no analysis errors in generated parts
    - Update generated code and golds
- Update boilerplate parsing to not throw when failing to instantiate an annotation, and instead severe-log. 
    - That way, analyzer plugin functionality that relies on boilerplate parsing doesn't emit an IDE error notification when typing an annotation (I noticed this while QAing this change): 
        ide-error-screenshot
     - Add regression test

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        - Verify in playground (tools/analyzer_plugin/playground/web/missing_required_props.dart) that:
            - the added `IgnoresSomeRequired` example doesn't warn for props that are ignored
            - adding/removing the `ignoreRequiredProps` argument doesn't cause IDE error notifications as you're typing
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Frontend Frameworks Design member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/builder/builder.dart [hunk 1]
```diff
@@ -253,6 +253,17 @@ class OverReactBuilder extends Builder {
...
 
+  static const _ignoreForFileCodes = {
+    // Needed when the component itself or some of its props are deprecated,
+    // or if deprecated types are referenced within the boilerplate.
+    'deprecated_member_use_from_same_package',
+    // These two are needed for the `?? null` workaround in prop accessors.
+    'unnecessary_null_in_if_null_operators',
+    'prefer_null_aware_operators',
+    // Needed to ignore the `requiredPropNamesToSkipValidation` checks.
+    'invalid_use_of_visible_for_overriding_member',
```
Comment (greglittlefield-wf) [inline lib/src/builder/builder.dart:264]: I added this ignore to fix my test setup, which validated there were no warnings in generated code

File: lib/src/builder/parsing/members/props_and_state_util.dart [hunk 2]
```diff
@@ -50,32 +50,54 @@ class _PropsStateStringHelpersImpl extends Object with PropsStateStringHelpers {
   _PropsStateStringHelpersImpl({required this.isProps});
 }
 
-final Expando<String> props_ignoreRequiredProps_source = Expando();
-
 /// Uses [InstantiatedMeta] to analyze [node] and determine the proper annotation.
 annotations.TypedMap getPropsOrStateAnnotation(bool isProps, AnnotatedNode node) {
```
Comment (greglittlefield-wf) [inline lib/src/builder/parsing/members/props_and_state_util.dart:54]: Whitespace-agnostic diff here is helpful

File: lib/src/builder/parsing/meta.dart [hunk 3]
```diff
@@ -102,31 +105,38 @@ class InstantiatedComponentMeta<TMeta> extends InstantiatedMeta<TMeta> {
   final Identifier? subtypeOfValue;
 
   InstantiatedComponentMeta._(
-      Annotation? metaNode, TMeta meta, List<Expression> unsupportedArguments, this.subtypeOfValue)
+      Annotation metaNode, TMeta meta, List<Expression> unsupportedArguments, this.subtypeOfValue)
       : super._(metaNode, meta, unsupportedArguments);
 
   static InstantiatedComponentMeta<T>? fromNode<T>(AnnotatedNode node) {
```
Comment (greglittlefield-wf) [inline lib/src/builder/parsing/meta.dart:111]: Whitespace-agnostic diff here is helpful

**Review by sydneyjodon-wk — APPROVED** [review]
+10

- [x] smoke tested in playground

**Review by sydneyjodon-wk — APPROVED** [review]
+10 refresh

**Pull request merged**